### PR TITLE
feat(op-reth): Canyon hardfork spec

### PR DESCRIPTION
**Pull request by clabby**
## Overview

> Note: Reopened version of #5491.
> This is the base PR in the Canyon HF stack. Precedence:
> #5502 
> -> `main`

Adds the `Canyon` hardfork to the `OP_GOERLI` and `BASE_GOERLI` chainspecs, as well as the `revm` config.

### CODE REVIEW

File: crates/primitives/src/chain/spec.rs [hunk 1]
```diff
@@ -1032,16 +1036,23 @@ impl ChainSpecBuilder {
...
         self.hardforks.insert(Hardfork::Regolith, ForkCondition::Timestamp(0));
         self
     }
 
+    /// Enable Canyon at genesis
+    #[cfg(feature = "optimism")]
+    pub fn canyon_activated(mut self) -> Self {
+        self = self.regolith_activated();
+        self.hardforks.insert(Hardfork::Shanghai, ForkCondition::Timestamp(0));
+        self.hardforks.insert(Hardfork::Canyon, ForkCondition::Timestamp(0));
```
Comment (rkrasiuk) [inline crates/primitives/src/chain/spec.rs:1052]:
```suggestion
        // Canyon also activates changes from L1's Shanghai hardfork
        self.hardforks.insert(Hardfork::Shanghai, ForkCondition::Timestamp(0));
        self.hardforks.insert(Hardfork::Canyon, ForkCondition::Timestamp(0));
```

File: crates/primitives/src/chain/spec.rs [hunk 2]
```diff
@@ -1976,6 +1987,7 @@ Post-merge hard forks (timestamp based):
 
     #[cfg(feature = "optimism")]
     #[test]
+    #[ignore]
```
Comment (rkrasiuk) [inline crates/primitives/src/chain/spec.rs:1990]: remove

File: crates/primitives/src/chain/spec.rs [hunk 3]
```diff
@@ -1989,8 +2001,47 @@ Post-merge hard forks (timestamp based):
...
+                    Head { number: 4061227, timestamp: 1699981200, ..Default::default() },
+                    ForkId { hash: ForkHash([0x7f, 0x4a, 0x72, 0x1f]), next: 0 },
+                ),
+            ],
+        );
+    }
+
+    #[cfg(feature = "optimism")]
+    #[test]
+    #[ignore]
```
Comment (rkrasiuk) [inline crates/primitives/src/chain/spec.rs:2021]: remove

**Review by mattsse — APPROVED** [review]
lgtm, pending @rkrasiuk

**Review by mattsse — APPROVED** [review]

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**